src/sentry/migrations/0029_discover_query_upgrade.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# Generated by Hand
+from __future__ import unicode_literals, print_function
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+from sentry.discover.models import DiscoverSavedQuery, DiscoverSavedQueryProject
+from sentry.api.event_search import SearchVisitor
```

Comment:
Neither of these can safely be imported here. They're very easily subject to breaking in the future and breaking this migration.

Ask @wedamija how to reference the models safely since I'm not sure how we do it in Django migrations, but we need to access them through the frozen state.

And for SearchVisitor, it looks like you're just referencing a couple constants. You should vendor those constants into this migration.